app/presenters/tree_builder_ae_class.rb

```diff
@@ -2,6 +2,23 @@ class TreeBuilderAeClass < TreeBuilder
   has_kids_for MiqAeClass, [:x_get_tree_class_kids, :type]
   has_kids_for MiqAeNamespace, [:x_get_tree_ns_kids, :type]
 
+  def initialize(name, type, sandbox, build = true, controller = nil)
```

Comment:
Cannot all this be resolved by passing a TreeNodeBuilder subclass as a constructor parameter?